ts/session/apis/snode_api/swarm_polling_config/SwarmPollingGroupConfig.ts

```diff
@@ -0,0 +1,250 @@
+    ) {
+      // delete any attachments in this conversation sent before that timestamp (in seconds)
+      const impactedMsgModels = await Data.getAllMessagesWithAttachmentsInConversationSentBefore({
+        deleteAttachBeforeSeconds: infos.deleteAttachBeforeSeconds,
+        conversationId: groupPk,
+      });
+      window.log.info(
+        `getAllMessagesWithAttachmentsInConversationSentBefore of ${ed25519Str(groupPk)} before ${infos.deleteAttachBeforeSeconds}: impactedMsgModelsIds `,
+        impactedMsgModels.map(m => m.id)
+      );
+
+      await destroyMessagesAndUpdateRedux(
+        impactedMsgModels.map(m => ({ conversationKey: groupPk, messageId: m.id }))
+      );
+
+      lastAppliedRemoveAttachmentSentBeforeSeconds.set(groupPk, infos.deleteAttachBeforeSeconds);
+    }
+  }
+  const membersWithPendingRemovals =
+    await MetaGroupWrapperActions.memberGetAllPendingRemovals(groupPk);
+  if (membersWithPendingRemovals.length) {
+    const group = await UserGroupsWrapperActions.getGroup(groupPk);
+    if (group && group.secretKey && !isEmpty(group.secretKey)) {
+      await GroupPendingRemovals.addJob({ groupPk });
+    }
+  }
+
+  const us = UserUtils.getOurPubKeyStrFromCache();
+  const usMember = await MetaGroupWrapperActions.memberGet(groupPk, us);
+  let keysAlreadyHaveAdmin = await MetaGroupWrapperActions.keysAdmin(groupPk);
+  const secretKeyInUserWrapper = (await UserGroupsWrapperActions.getGroup(groupPk))?.secretKey;
+
+  // load admin keys if needed
+  if (
+    usMember &&
+    secretKeyInUserWrapper &&
+    !isEmpty(secretKeyInUserWrapper) &&
+    !keysAlreadyHaveAdmin
+  ) {
+    try {
+      await MetaGroupWrapperActions.loadAdminKeys(groupPk, secretKeyInUserWrapper);
+      keysAlreadyHaveAdmin = await MetaGroupWrapperActions.keysAdmin(groupPk);
+    } catch (e) {
+      window.log.warn(
+        `tried to update our adminKeys/state for group ${ed25519Str(groupPk)} but failed with, ${e.message}`
+      );
+    }
+  }
+  // mark ourselves as accepting the invite if needed
+  if (usMember?.memberStatus === 'INVITE_SENT' && keysAlreadyHaveAdmin) {
+    await MetaGroupWrapperActions.memberSetAccepted(groupPk, us);
+  }
+  // mark ourselves as accepting the promotion if needed
+  if (usMember?.memberStatus === 'PROMOTION_SENT' && keysAlreadyHaveAdmin) {
+    await MetaGroupWrapperActions.memberSetPromotionAccepted(groupPk, us);
+  }
+  // this won't do anything if there is no need for a sync, so we can safely plan one
+  await GroupSync.queueNewJobIfNeeded(groupPk);
+
+  const convo = ConvoHub.use().get(groupPk);
+  const refreshedInfos = await MetaGroupWrapperActions.infoGet(groupPk);
+
+  if (convo) {
+    let changes = false;
+    if (refreshedInfos.name !== convo.get('displayNameInProfile')) {
+      convo.set({ displayNameInProfile: refreshedInfos.name || undefined });
+      changes = true;
+    }
+    const expectedMode = refreshedInfos.expirySeconds ? 'deleteAfterSend' : 'off';
```

Comment:
Is this the correct name or a typo for `expirationMode`?

Reply:
well, expected because we then check for it to be what we expect.
I can rename it to expectedExpirationMode if you want.